internal/controller/styra/system_controller.go

```diff
@@ -1090,6 +1099,19 @@ func (r *SystemReconciler) specToSystemConfig(system *v1beta1.System) *styra.Sys
 		ReadOnly: r.Config.ReadOnly,
 	}
 
+	enableDeltaBundles := true
+	if r.Config.EnableDeltaBundlesDefault != nil {
+		enableDeltaBundles = *r.Config.EnableDeltaBundlesDefault
+	}
+	if system.Spec.EnableDeltaBundles != nil {
+		enableDeltaBundles = *system.Spec.EnableDeltaBundles
+	}
+
+	bundleDownloadConfig := &styra.BundleDownloadConfig{
+		DeltaBundles: enableDeltaBundles,
+	}
+	cfg.BundleDownload = bundleDownloadConfig
+
```

Comment:
In my opinion there is not reason to create a variable that is only used once. 

```suggestion
	cfg.BundleDownload = &styra.BundleDownloadConfig{
		DeltaBundles: enableDeltaBundles,
	}
```